[bugfix] compile error

1. gcc version:  riscv64-unknown-linux-gnu-gcc (g04696df0963) 14.2.0
2. compile cmd:
```
make ARCH=riscv CROSS_COMPILE=riscv64-unknown-linux-gnu- defconfig
make ARCH=riscv CROSS_COMPILE=riscv64-unknown-linux-gnu- -j$(nproc) bindeb-pkg
```
3. error log:
```
drivers/gpu/drm/img-rogue/physmem_hostmem.c:69:9: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [-Werror=designated-init]
   69 |         HostMemCpuPAddrToDevPAddr,
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~
drivers/gpu/drm/img-rogue/physmem_hostmem.c:69:9: note: (near initialization for 'gsHostMemDevPhysHeapFuncs')
drivers/gpu/drm/img-rogue/physmem_hostmem.c:71:9: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [-Werror=designated-init]
   71 |         HostMemDevPAddrToCpuPAddr,
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~
drivers/gpu/drm/img-rogue/physmem_hostmem.c:71:9: note: (near initialization for 'gsHostMemDevPhysHeapFuncs')
cc1: some warnings being treated as errors
```

```
drivers/gpu/drm/img-rogue/physmem_lma.c:1846:9: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [-Werror=designated-init]
 1846 |         &PMRLockSysPhysAddressesLocalMem,
      |         ^
drivers/gpu/drm/img-rogue/physmem_lma.c:1846:9: note: (near initialization for '_sPMRLMAFuncTab')
drivers/gpu/drm/img-rogue/physmem_lma.c:1846:9: error: initialization of 'void (*)(void)' from incompatible pointer type 'PVRSRV_ERROR (*)(void *)' {aka 'enum PVRSRV_ERROR_TAG (*)(void *)'} [-Wincompatible-pointer-types]
drivers/gpu/drm/img-rogue/physmem_lma.c:1846:9: note: (near initialization for '_sPMRLMAFuncTab.pfnGetPMRFactoryLock')
drivers/gpu/drm/img-rogue/physmem_lma.c:1848:9: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [-Werror=designated-init]
 1848 |         &PMRUnlockSysPhysAddressesLocalMem,
      |         ^
drivers/gpu/drm/img-rogue/physmem_lma.c:1848:9: note: (near initialization for '_sPMRLMAFuncTab')
drivers/gpu/drm/img-rogue/physmem_lma.c:1848:9: error: initialization of 'PVRSRV_ERROR (*)(void *, const PMR *, IMG_UINT64,  IMG_UINT32,  IMG_UINT32 *, IMG_UINT32,  IMG_UINT32 *)' {aka 'enum PVRSRV_ERROR_TAG (*)(void *, const struct _PMR_ *, long long unsigned int,  unsigned int,  unsigned int *, unsigned int,  unsigned int *)'} from incompatible pointer type 'PVRSRV_ERROR (*)(void *)' {aka 'enum PVRSRV_ERROR_TAG (*)(void *)'} [-Wincompatible-pointer-types]
drivers/gpu/drm/img-rogue/physmem_lma.c:1848:9: note: (near initialization for '_sPMRLMAFuncTab.pfnChangeSparseMemCPUMap')
drivers/gpu/drm/img-rogue/physmem_lma.c:1850:9: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [-Werror=designated-init]
 1850 |         &PMRSysPhysAddrLocalMem,
      |         ^
drivers/gpu/drm/img-rogue/physmem_lma.c:1850:9: note: (near initialization for '_sPMRLMAFuncTab')
drivers/gpu/drm/img-rogue/physmem_lma.c:1850:9: error: initialization of 'void (*)(void)' from incompatible pointer type 'PVRSRV_ERROR (*)(void *, IMG_UINT32,  IMG_UINT32,  IMG_DEVMEM_OFFSET_T *, IMG_BOOL *, IMG_DEV_PHYADDR *)' {aka 'enum PVRSRV_ERROR_TAG (*)(void *, unsigned int,  unsigned int,  long long unsigned int *, enum tag_img_bool *, IMG_DEV_PHYADDR *)'} [-Wincompatible-pointer-types]
drivers/gpu/drm/img-rogue/physmem_lma.c:1850:9: note: (near initialization for '_sPMRLMAFuncTab.pfnReleasePMRFactoryLock')
drivers/gpu/drm/img-rogue/physmem_lma.c:1852:9: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [-Werror=designated-init]
 1852 |         &PMRAcquireKernelMappingDataLocalMem,
      |         ^
drivers/gpu/drm/img-rogue/physmem_lma.c:1852:9: note: (near initialization for '_sPMRLMAFuncTab')
drivers/gpu/drm/img-rogue/physmem_lma.c:1852:9: error: initialization of 'PVRSRV_ERROR (*)(void *, IMG_UINT32,  IMG_UINT32,  IMG_DEVMEM_OFFSET_T *, IMG_BOOL *, IMG_DEV_PHYADDR *)' {aka 'enum PVRSRV_ERROR_TAG (*)(void *, unsigned int,  unsigned int,  long long unsigned int *, enum tag_img_bool *, IMG_DEV_PHYADDR *)'} from incompatible pointer type 'PVRSRV_ERROR (*)(void *, size_t,  size_t,  void **, void **, PMR_FLAGS_T)' {aka 'enum PVRSRV_ERROR_TAG (*)(void *, long unsigned int,  long unsigned int,  void **, void **, long long unsigned int)'} [-Wincompatible-pointer-types]
drivers/gpu/drm/img-rogue/physmem_lma.c:1852:9: note: (near initialization for '_sPMRLMAFuncTab.pfnDevPhysAddr')
drivers/gpu/drm/img-rogue/physmem_lma.c:1854:9: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [-Werror=designated-init]
 1854 |         &PMRReleaseKernelMappingDataLocalMem,
      |         ^
drivers/gpu/drm/img-rogue/physmem_lma.c:1854:9: note: (near initialization for '_sPMRLMAFuncTab')
drivers/gpu/drm/img-rogue/physmem_lma.c:1856:9: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [-Werror=designated-init]
 1856 |         &PMRReadBytesLocalMem,
      |         ^
drivers/gpu/drm/img-rogue/physmem_lma.c:1856:9: note: (near initialization for '_sPMRLMAFuncTab')
drivers/gpu/drm/img-rogue/physmem_lma.c:1856:9: error: initialization of 'PVRSRV_ERROR (*)(void *, size_t,  size_t,  void **, void **, PMR_FLAGS_T)' {aka 'enum PVRSRV_ERROR_TAG (*)(void *, long unsigned int,  long unsigned int,  void **, void **, long long unsigned int)'} from incompatible pointer type 'PVRSRV_ERROR (*)(void *, IMG_DEVMEM_OFFSET_T,  IMG_UINT8 *, size_t,  size_t *)' {aka 'enum PVRSRV_ERROR_TAG (*)(void *, long long unsigned int,  unsigned char *, long unsigned int,  long unsigned int *)'} [-Wincompatible-pointer-types]
drivers/gpu/drm/img-rogue/physmem_lma.c:1856:9: note: (near initialization for '_sPMRLMAFuncTab.pfnAcquireKernelMappingData')
drivers/gpu/drm/img-rogue/physmem_lma.c:1858:9: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [-Werror=designated-init]
 1858 |         &PMRWriteBytesLocalMem,
      |         ^
drivers/gpu/drm/img-rogue/physmem_lma.c:1858:9: note: (near initialization for '_sPMRLMAFuncTab')
drivers/gpu/drm/img-rogue/physmem_lma.c:1858:9: error: initialization of 'PVRSRV_ERROR (*)(void *, const PMR *, IMG_UINT32,  IMG_UINT32 *, IMG_UINT32,  IMG_UINT32 *, IMG_UINT32)' {aka 'enum PVRSRV_ERROR_TAG (*)(void *, const struct _PMR_ *, unsigned int,  unsigned int *, unsigned int,  unsigned int *, unsigned int)'} from incompatible pointer type 'PVRSRV_ERROR (*)(void *, IMG_DEVMEM_OFFSET_T,  IMG_UINT8 *, size_t,  size_t *)' {aka 'enum PVRSRV_ERROR_TAG (*)(void *, long long unsigned int,  unsigned char *, long unsigned int,  long unsigned int *)'} [-Wincompatible-pointer-types]
drivers/gpu/drm/img-rogue/physmem_lma.c:1858:9: note: (near initialization for '_sPMRLMAFuncTab.pfnChangeSparseMem')
In file included from ./include/uapi/linux/posix_types.h:5,
                 from ./include/uapi/linux/types.h:14,
                 from ./include/linux/types.h:6,
                 from drivers/gpu/drm/img-rogue/img_types.h:69,
                 from drivers/gpu/drm/img-rogue/physmem_lma.c:45:
./include/linux/stddef.h:8:14: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [-Werror=designated-init]
    8 | #define NULL ((void *)0)
      |              ^
drivers/gpu/drm/img-rogue/physmem_lma.c:1860:9: note: in expansion of macro 'NULL'
 1860 |         NULL,
      |         ^~~~
./include/linux/stddef.h:8:14: note: (near initialization for '_sPMRLMAFuncTab')
    8 | #define NULL ((void *)0)
      |              ^
drivers/gpu/drm/img-rogue/physmem_lma.c:1860:9: note: in expansion of macro 'NULL'
 1860 |         NULL,
      |         ^~~~
./include/linux/stddef.h:8:14: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [-Werror=designated-init]
    8 | #define NULL ((void *)0)
      |              ^
drivers/gpu/drm/img-rogue/physmem_lma.c:1862:9: note: in expansion of macro 'NULL'
 1862 |         NULL,
      |         ^~~~
./include/linux/stddef.h:8:14: note: (near initialization for '_sPMRLMAFuncTab')
    8 | #define NULL ((void *)0)
      |              ^
drivers/gpu/drm/img-rogue/physmem_lma.c:1862:9: note: in expansion of macro 'NULL'
 1862 |         NULL,
      |         ^~~~
drivers/gpu/drm/img-rogue/physmem_lma.c:1864:9: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [-Werror=designated-init]
 1864 |         &PMRChangeSparseMemLocalMem,
      |         ^
drivers/gpu/drm/img-rogue/physmem_lma.c:1864:9: note: (near initialization for '_sPMRLMAFuncTab')
drivers/gpu/drm/img-rogue/physmem_lma.c:1864:9: error: initialization of 'PVRSRV_ERROR (*)(void *, IMG_DEVMEM_OFFSET_T,  IMG_UINT8 *, size_t,  size_t *)' {aka 'enum PVRSRV_ERROR_TAG (*)(void *, long long unsigned int,  unsigned char *, long unsigned int,  long unsigned int *)'} from incompatible pointer type 'PVRSRV_ERROR (*)(void *, const PMR *, IMG_UINT32,  IMG_UINT32 *, IMG_UINT32,  IMG_UINT32 *, IMG_UINT32)' {aka 'enum PVRSRV_ERROR_TAG (*)(void *, const struct _PMR_ *, unsigned int,  unsigned int *, unsigned int,  unsigned int *, unsigned int)'} [-Wincompatible-pointer-types]
drivers/gpu/drm/img-rogue/physmem_lma.c:1864:9: note: (near initialization for '_sPMRLMAFuncTab.pfnReadBytes')
drivers/gpu/drm/img-rogue/physmem_lma.c:1866:9: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [-Werror=designated-init]
 1866 |         &PMRChangeSparseMemCPUMapLocalMem,
      |         ^
drivers/gpu/drm/img-rogue/physmem_lma.c:1866:9: note: (near initialization for '_sPMRLMAFuncTab')
drivers/gpu/drm/img-rogue/physmem_lma.c:1866:9: error: initialization of 'PVRSRV_ERROR (*)(void *)' {aka 'enum PVRSRV_ERROR_TAG (*)(void *)'} from incompatible pointer type 'PVRSRV_ERROR (*)(void *, const PMR *, IMG_UINT64,  IMG_UINT32,  IMG_UINT32 *, IMG_UINT32,  IMG_UINT32 *)' {aka 'enum PVRSRV_ERROR_TAG (*)(void *, const struct _PMR_ *, long long unsigned int,  unsigned int,  unsigned int *, unsigned int,  unsigned int *)'} [-Wincompatible-pointer-types]
drivers/gpu/drm/img-rogue/physmem_lma.c:1866:9: note: (near initialization for '_sPMRLMAFuncTab.pfnUnlockPhysAddresses')
./include/linux/stddef.h:8:14: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [-Werror=designated-init]
    8 | #define NULL ((void *)0)
      |              ^
drivers/gpu/drm/img-rogue/physmem_lma.c:1868:9: note: in expansion of macro 'NULL'
 1868 |         NULL,
      |         ^~~~
./include/linux/stddef.h:8:14: note: (near initialization for '_sPMRLMAFuncTab')
    8 | #define NULL ((void *)0)
      |              ^
drivers/gpu/drm/img-rogue/physmem_lma.c:1868:9: note: in expansion of macro 'NULL'
 1868 |         NULL,
      |         ^~~~
drivers/gpu/drm/img-rogue/physmem_lma.c:1870:9: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [-Werror=designated-init]
 1870 |         &PMRFinalizeLocalMem
      |         ^
drivers/gpu/drm/img-rogue/physmem_lma.c:1870:9: note: (near initialization for '_sPMRLMAFuncTab')
cc1: some warnings being treated as errors
make[6]: *** [scripts/Makefile.build:243：drivers/gpu/drm/img-rogue/physmem_lma.o] 错误 1
make[5]: *** [scripts/Makefile.build:480：drivers/gpu/drm/img-rogue] 错误 2
make[4]: *** [scripts/Makefile.build:480：drivers/gpu/drm] 错误 2
make[3]: *** [scripts/Makefile.build:480：drivers/gpu] 错误 2
make[2]: *** [scripts/Makefile.build:480：drivers] 错误 2
make[1]: *** [/mnt/hdd_1/wangjia/RVCK/rvck/Makefile:1924：.] 错误 2
make: *** [Makefile:234：__sub-make] 错误 2
```

```
drivers/gpu/drm/img-rogue/vmm_type_stub.c:78:17: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [-Werror=designated-init]
   78 |                 &StubVMMMapDevPhysHeap,
      |                 ^
drivers/gpu/drm/img-rogue/vmm_type_stub.c:78:17: note: (near initialization for 'gsStubVmmPvz.sClientFuncTab')
drivers/gpu/drm/img-rogue/vmm_type_stub.c:78:17: error: initialization of 'PVRSRV_ERROR (*)(IMG_UINT32,  IMG_UINT32)' {aka 'enum PVRSRV_ERROR_TAG (*)(unsigned int,  unsigned int)'} from incompatible pointer type 'PVRSRV_ERROR (*)(IMG_UINT32,  IMG_UINT32,  IMG_UINT64,  IMG_UINT64)' {aka 'enum PVRSRV_ERROR_TAG (*)(unsigned int,  unsigned int,  long long unsigned int,  long long unsigned int)'} [-Wincompatible-pointer-types]
drivers/gpu/drm/img-rogue/vmm_type_stub.c:78:17: note: (near initialization for 'gsStubVmmPvz.sClientFuncTab.pfnUnmapDevPhysHeap')
drivers/gpu/drm/img-rogue/vmm_type_stub.c:81:17: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [-Werror=designated-init]
   81 |                 &StubVMMUnmapDevPhysHeap
      |                 ^
drivers/gpu/drm/img-rogue/vmm_type_stub.c:81:17: note: (near initialization for 'gsStubVmmPvz.sClientFuncTab')
drivers/gpu/drm/img-rogue/vmm_type_stub.c:81:17: error: initialization of 'PVRSRV_ERROR (*)(IMG_UINT32,  IMG_UINT32,  IMG_UINT64,  IMG_UINT64)' {aka 'enum PVRSRV_ERROR_TAG (*)(unsigned int,  unsigned int,  long long unsigned int,  long long unsigned int)'} from incompatible pointer type 'PVRSRV_ERROR (*)(IMG_UINT32,  IMG_UINT32)' {aka 'enum PVRSRV_ERROR_TAG (*)(unsigned int,  unsigned int)'} [-Wincompatible-pointer-types]
drivers/gpu/drm/img-rogue/vmm_type_stub.c:81:17: note: (near initialization for 'gsStubVmmPvz.sClientFuncTab.pfnMapDevPhysHeap')
drivers/gpu/drm/img-rogue/vmm_type_stub.c:86:17: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [-Werror=designated-init]
   86 |                 &PvzServerMapDevPhysHeap,
      |                 ^
drivers/gpu/drm/img-rogue/vmm_type_stub.c:86:17: note: (near initialization for 'gsStubVmmPvz.sServerFuncTab')
drivers/gpu/drm/img-rogue/vmm_type_stub.c:86:17: error: initialization of 'PVRSRV_ERROR (*)(IMG_UINT32,  IMG_UINT32,  IMG_UINT32)' {aka 'enum PVRSRV_ERROR_TAG (*)(unsigned int,  unsigned int,  unsigned int)'} from incompatible pointer type 'PVRSRV_ERROR (*)(IMG_UINT32,  IMG_UINT32,  IMG_UINT32,  IMG_UINT64,  IMG_UINT64)' {aka 'enum PVRSRV_ERROR_TAG (*)(unsigned int,  unsigned int,  unsigned int,  long long unsigned int,  long long unsigned int)'} [-Wincompatible-pointer-types]
drivers/gpu/drm/img-rogue/vmm_type_stub.c:86:17: note: (near initialization for 'gsStubVmmPvz.sServerFuncTab.pfnUnmapDevPhysHeap')
drivers/gpu/drm/img-rogue/vmm_type_stub.c:89:17: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [-Werror=designated-init]
   89 |                 &PvzServerUnmapDevPhysHeap
      |                 ^
drivers/gpu/drm/img-rogue/vmm_type_stub.c:89:17: note: (near initialization for 'gsStubVmmPvz.sServerFuncTab')
drivers/gpu/drm/img-rogue/vmm_type_stub.c:89:17: error: initialization of 'PVRSRV_ERROR (*)(IMG_UINT32,  IMG_UINT32,  IMG_UINT32,  IMG_UINT64,  IMG_UINT64)' {aka 'enum PVRSRV_ERROR_TAG (*)(unsigned int,  unsigned int,  unsigned int,  long long unsigned int,  long long unsigned int)'} from incompatible pointer type 'PVRSRV_ERROR (*)(IMG_UINT32,  IMG_UINT32,  IMG_UINT32)' {aka 'enum PVRSRV_ERROR_TAG (*)(unsigned int,  unsigned int,  unsigned int)'} [-Wincompatible-pointer-types]
drivers/gpu/drm/img-rogue/vmm_type_stub.c:89:17: note: (near initialization for 'gsStubVmmPvz.sServerFuncTab.pfnMapDevPhysHeap')
drivers/gpu/drm/img-rogue/vmm_type_stub.c:94:17: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [-Werror=designated-init]
   94 |                 &PvzServerOnVmOnline,
      |                 ^
drivers/gpu/drm/img-rogue/vmm_type_stub.c:94:17: note: (near initialization for 'gsStubVmmPvz.sVmmFuncTab')
drivers/gpu/drm/img-rogue/vmm_type_stub.c:97:17: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [-Werror=designated-init]
   97 |                 &PvzServerOnVmOffline,
      |                 ^
drivers/gpu/drm/img-rogue/vmm_type_stub.c:97:17: note: (near initialization for 'gsStubVmmPvz.sVmmFuncTab')
drivers/gpu/drm/img-rogue/vmm_type_stub.c:97:17: error: initialization of 'PVRSRV_ERROR (*)(VMM_CONF_PARAM,  IMG_UINT32)' {aka 'enum PVRSRV_ERROR_TAG (*)(enum _VMM_CONF_PARAM_,  unsigned int)'} from incompatible pointer type 'PVRSRV_ERROR (*)(IMG_UINT32)' {aka 'enum PVRSRV_ERROR_TAG (*)(unsigned int)'} [-Wincompatible-pointer-types]
drivers/gpu/drm/img-rogue/vmm_type_stub.c:97:17: note: (near initialization for 'gsStubVmmPvz.sVmmFuncTab.pfnVMMConfigure')
drivers/gpu/drm/img-rogue/vmm_type_stub.c:100:17: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [-Werror=designated-init]
  100 |                 &PvzServerVMMConfigure
      |                 ^
drivers/gpu/drm/img-rogue/vmm_type_stub.c:100:17: note: (near initialization for 'gsStubVmmPvz.sVmmFuncTab')
drivers/gpu/drm/img-rogue/vmm_type_stub.c:100:17: error: initialization of 'PVRSRV_ERROR (*)(IMG_UINT32)' {aka 'enum PVRSRV_ERROR_TAG (*)(unsigned int)'} from incompatible pointer type 'PVRSRV_ERROR (*)(VMM_CONF_PARAM,  IMG_UINT32)' {aka 'enum PVRSRV_ERROR_TAG (*)(enum _VMM_CONF_PARAM_,  unsigned int)'} [-Wincompatible-pointer-types]
drivers/gpu/drm/img-rogue/vmm_type_stub.c:100:17: note: (near initialization for 'gsStubVmmPvz.sVmmFuncTab.pfnOnVmOnline')
cc1: some warnings being treated as errors
make[10]: *** [scripts/Makefile.build:243：drivers/gpu/drm/img-rogue/vmm_type_stub.o] 错误 1
make[10]: *** 正在等待未完成的任务....
  CC [M]  drivers/gpu/drm/drm_bridge_connector.o
  CC [M]  drivers/gpu/drm/drm_crtc_helper.o
  CC [M]  drivers/gpu/drm/drm_damage_helper.o
drivers/gpu/drm/img-rogue/sysconfig.c:154:9: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [-Werror=designated-init]
  154 |         UMAPhysHeapCpuPAddrToDevPAddr,
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
drivers/gpu/drm/img-rogue/sysconfig.c:154:9: note: (near initialization for 'gsPhysHeapFuncs')
drivers/gpu/drm/img-rogue/sysconfig.c:156:9: error: positional initialization of field in 'struct' declared with 'designated_init' attribute [-Werror=designated-init]
  156 |         UMAPhysHeapDevPAddrToCpuPAddr,
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
drivers/gpu/drm/img-rogue/sysconfig.c:156:9: note: (near initialization for 'gsPhysHeapFuncs')
  CC [M]  drivers/gpu/drm/drm_encoder_slave.o
  CC [M]  drivers/gpu/drm/drm_flip_work.o
cc1: some warnings being treated as errors
make[10]: *** [scripts/Makefile.build:243：drivers/gpu/drm/img-rogue/sysconfig.o] 错误 1
```

--- a/drivers/gpu/drm/img-rogue/physmem_hostmem.c
+++ b/drivers/gpu/drm/img-rogue/physmem_hostmem.c
@@ -66,9 +66,9 @@ static void HostMemDevPAddrToCpuPAddr(IMG_HANDLE hPrivData,
 static PHYS_HEAP_FUNCTIONS gsHostMemDevPhysHeapFuncs =
 {
 	/* pfnCpuPAddrToDevPAddr */
-	HostMemCpuPAddrToDevPAddr,
+	.pfnCpuPAddrToDevPAddr = HostMemCpuPAddrToDevPAddr,
 	/* pfnDevPAddrToCpuPAddr */
-	HostMemDevPAddrToCpuPAddr,
+	.pfnDevPAddrToCpuPAddr = HostMemDevPAddrToCpuPAddr
 };
 
 static PVRSRV_DEVICE_CONFIG gsHostMemDevConfig[];

--- a/drivers/gpu/drm/img-rogue/physmem_lma.c
+++ b/drivers/gpu/drm/img-rogue/physmem_lma.c
@@ -1843,31 +1843,31 @@ PVRSRV_ERROR PMRChangeSparseMemCPUMapLocalMem(PMR_IMPL_PRIVDATA pPriv,
 
 static PMR_IMPL_FUNCTAB _sPMRLMAFuncTab = {
 	/* pfnLockPhysAddresses */
-	&PMRLockSysPhysAddressesLocalMem,
+	.pfnLockPhysAddresses = &PMRLockSysPhysAddressesLocalMem,
 	/* pfnUnlockPhysAddresses */
-	&PMRUnlockSysPhysAddressesLocalMem,
+	.pfnUnlockPhysAddresses = &PMRUnlockSysPhysAddressesLocalMem,
 	/* pfnDevPhysAddr */
-	&PMRSysPhysAddrLocalMem,
+	.pfnDevPhysAddr = &PMRSysPhysAddrLocalMem,
 	/* pfnAcquireKernelMappingData */
-	&PMRAcquireKernelMappingDataLocalMem,
+	.pfnAcquireKernelMappingData = &PMRAcquireKernelMappingDataLocalMem,
 	/* pfnReleaseKernelMappingData */
-	&PMRReleaseKernelMappingDataLocalMem,
+	.pfnReleaseKernelMappingData = &PMRReleaseKernelMappingDataLocalMem,
 	/* pfnReadBytes */
-	&PMRReadBytesLocalMem,
+	.pfnReadBytes = &PMRReadBytesLocalMem,
 	/* pfnWriteBytes */
-	&PMRWriteBytesLocalMem,
+	.pfnWriteBytes = &PMRWriteBytesLocalMem,
 	/* pfnUnpinMem */
-	NULL,
+	.pfnUnpinMem = NULL,
 	/* pfnPinMem */
-	NULL,
+	.pfnPinMem = NULL,
 	/* pfnChangeSparseMem*/
-	&PMRChangeSparseMemLocalMem,
+	.pfnChangeSparseMem = &PMRChangeSparseMemLocalMem,
 	/* pfnChangeSparseMemCPUMap */
-	&PMRChangeSparseMemCPUMapLocalMem,
+	.pfnChangeSparseMemCPUMap = &PMRChangeSparseMemCPUMapLocalMem,
 	/* pfnMMap */
-	NULL,
+	.pfnMMap = NULL,
 	/* pfnFinalize */
-	&PMRFinalizeLocalMem
+	.pfnFinalize = &PMRFinalizeLocalMem
 };
 
 PVRSRV_ERROR

--- a/drivers/gpu/drm/img-rogue/sysconfig.c
+++ b/drivers/gpu/drm/img-rogue/sysconfig.c
@@ -151,9 +151,9 @@ void UMAPhysHeapDevPAddrToCpuPAddr(IMG_HANDLE hPrivData,
 static PHYS_HEAP_FUNCTIONS gsPhysHeapFuncs =
 {
 	/* pfnCpuPAddrToDevPAddr */
-	UMAPhysHeapCpuPAddrToDevPAddr,
+	.pfnCpuPAddrToDevPAddr = UMAPhysHeapCpuPAddrToDevPAddr,
 	/* pfnDevPAddrToCpuPAddr */
-	UMAPhysHeapDevPAddrToCpuPAddr,
+	.pfnDevPAddrToCpuPAddr = UMAPhysHeapDevPAddrToCpuPAddr,
 };
 
 static PVRSRV_ERROR PhysHeapsCreate(PHYS_HEAP_CONFIG **ppasPhysHeapsOut,

--- a/drivers/gpu/drm/img-rogue/vmm_type_stub.c
+++ b/drivers/gpu/drm/img-rogue/vmm_type_stub.c
@@ -75,29 +75,29 @@ static VMM_PVZ_CONNECTION gsStubVmmPvz =
 {
 	.sClientFuncTab = {
 		/* pfnMapDevPhysHeap */
-		&StubVMMMapDevPhysHeap,
+		.pfnMapDevPhysHeap = &StubVMMMapDevPhysHeap,
 
 		/* pfnUnmapDevPhysHeap */
-		&StubVMMUnmapDevPhysHeap
+		.pfnUnmapDevPhysHeap = &StubVMMUnmapDevPhysHeap
 	},
 
 	.sServerFuncTab = {
 		/* pfnMapDevPhysHeap */
-		&PvzServerMapDevPhysHeap,
+		.pfnMapDevPhysHeap = &PvzServerMapDevPhysHeap,
 
 		/* pfnUnmapDevPhysHeap */
-		&PvzServerUnmapDevPhysHeap
+		.pfnUnmapDevPhysHeap = &PvzServerUnmapDevPhysHeap
 	},
 
 	.sVmmFuncTab = {
 		/* pfnOnVmOnline */
-		&PvzServerOnVmOnline,
+		.pfnOnVmOnline = &PvzServerOnVmOnline,
 
 		/* pfnOnVmOffline */
-		&PvzServerOnVmOffline,
+		.pfnOnVmOffline = &PvzServerOnVmOffline,
 
 		/* pfnVMMConfigure */
-		&PvzServerVMMConfigure
+		.pfnVMMConfigure = &PvzServerVMMConfigure
 	}
 };
 

